### PR TITLE
Fix: Add InvalidCustomImage, InvalidArch, and InvalidOs image build errors

### PR DESCRIPTION
**Pull request by jsun-m**
During n session where user-client connects to gateway to listen for logs from a container, there is a likelihood for the container to fail before it's able to open a port to allow gateway to connect to it. Because of this, the gateway continues to look for the connection before timing out for 5 minutes. 

This makes the gateway watch for the container exit code before it connects as well as utilize the exit code for useful information that can be forwarded to user.

Also added InvalidArch as well as InvalidOs to make sure user is using a custom image for linux/amd64

### CODE REVIEW

**Review by nickpetrovic — CHANGES_REQUESTED** [review]

File: pkg/worker/image.go [hunk 1]
```diff
@@ -215,19 +216,59 @@ func (c *ImageClient) Cleanup() error {
...
+
+	args = append(args, c.inspectArgs(creds)...)
+	cmd := exec.CommandContext(ctx, c.pullCommand, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	output, err := exec.CommandContext(ctx, c.pullCommand, args...).Output()
+	if err != nil {
+		return &types.ExitCodeError{
+			ExitCode: types.WorkerContainerExitCodeInvalidCustomImage,
```
[the remark this answers is not in the document]
Comment (jsun-m) [reply pkg/worker/image.go:230]: We can. However, I'm just expanding on the exitcode value we already have without adding anything new.

File: pkg/worker/image.go [hunk 2]
```diff
@@ -241,7 +282,7 @@ func (c *ImageClient) PullAndArchiveImage(ctx context.Context, sourceImage strin
 
 	status, err := runc.Monitor.Wait(cmd, ec)
 	if err == nil && status != 0 {
-		return fmt.Errorf("unable to copy image: %v", cmd.String())
+		return fmt.Errorf("unable to pull image: %v", status)
```
[the remark this answers is not in the document]
Comment (jsun-m) [reply pkg/worker/image.go:285]: not sure why I did this. I think i was debugging. Gonna revert

File: pkg/worker/image.go [hunk 3]
```diff
@@ -286,6 +327,30 @@ func (c *ImageClient) args(creds string) (out []string) {
...
+	if !c.config.ImageService.EnableTLS {
+		out = append(out, []string{"--tls-verify=false"}...)
+	}
+
+	if c.debug {
+		out = append(out, "--debug")
+	}
+
+	return out
+}
```
[the remark this answers is not in the document]
Comment (jsun-m) [reply pkg/worker/image.go:352]: yea these are unique flags to the inspect command. For some reason they had different names for different flags between `copy` and `inspect`

File: pkg/worker/image.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickpetrovic) [reply pkg/worker/image.go:285]: Sorry didn't realize myself that status was an int - didn't mean to be snarky. So yeah this doesn't show the command.

File: pkg/repository/container_redis.go [hunk 4]
```diff
@@ -197,8 +197,8 @@ func (cr *ContainerRedisRepository) SetWorkerAddress(containerId string, addr st
 	return cr.rdb.Set(context.TODO(), common.RedisKeys.SchedulerWorkerAddress(containerId), addr, 0).Err()
 }
 
-func (cr *ContainerRedisRepository) GetWorkerAddress(containerId string) (string, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+func (cr *ContainerRedisRepository) GetWorkerAddress(ctx context.Context, containerId string) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
```
[the remark this answers is not in the document]
Comment (jsun-m) [reply pkg/repository/container_redis.go:201]: do you mean that the ctx, cancel here overrides the ctx value?

[the remark this answers is not in the document]
Comment (jsun-m) [reply pkg/repository/container_redis.go]: I just renamed the ctx value so its more clear

[the remark this answers is not in the document]
Comment (nickpetrovic) [reply pkg/repository/container_redis.go]: No, overriding the ctx is fine and should be done, but int he for loop below this when the context is done, we're saying its a timeout which may not be true if the parent context is canceled.

[the remark this answers is not in the document]
Comment (jsun-m) [reply pkg/repository/container_redis.go]: oh got it. i get what you mean

**Review by nickpetrovic — APPROVED** [review]

**Pull request merged**